CI

AlexanderKiselyov opened the pull request.

donebd committed:
Start building architecture

donebd committed:
Menu fragment can be started =)

donebd committed:
Search fragment can be started =)

AirBurn0 committed:
minor layout fixes

AirBurn0 committed:
item rework

AirBurn0 committed:
add navbar

donebd committed:
replace dto by real obj in repo

AirBurn0 committed:
search bar

AirBurn0 committed:
major UI rework

AirBurn0 committed:
mobile_nav

AirBurn0 committed:
add preview and modify&double lib/ext item

Anilochka committed:
Repo Mock (#10)

* Added Repository mock and simple quests-generator class

* Changes findByName

* Deleted unused files

* Changed list to map

donebd committed:
WIP WIP

donebd committed:
minor changes

donebd committed:
Переместил папочки в нужное место

Commit:
Add QuestContent, Walkthrough models

donebd committed:
Допилил чуть переходы

donebd committed:
resolve branch conflict

donebd committed:
Ability to remove quest, delete external_preview layout

donebd committed:
допилил ui логику privewFragment'а

AirBurn0 committed:
format

AirBurn0 committed:
fix typo

levs2001 committed:
Merge pull request #16

* Start

* Content from json

* Add doc comments to Walkthrough and QuestContent

* Merge pull request #15

* Merged

* Deleted unnes

* Ignore .idea

* Renaming comments

* Merge branch 'ui_dev'

* Merged

donebd committed:
уберите это из ресурсов

donebd committed:
delete backstack dispatcher

Commit:
Add hobbit quest

Commit:
Add meta for hobbit

Alexandra-Tkachenko committed:
Fix text quest_information.json

Alexandra-Tkachenko committed:
Fix text hobbit.json

donebd committed:
resolve branch conflict

levs2001 committed:
Quest metainfo json (#19)

* Start

* Deleted unnes

* Ignore .idea

* Renaming comments

* Merged

* Metas

* removed uneccessary comments

* Removed unnecces

* resolve branch conflict

Co-authored-by: done_bd <dinil.bd@mail.ru>

donebd committed:
delete unused

AirBurn0 committed:
Quest UI (#21)

* Fix option menu (без фикса навигации)

* title change

* Search realisation in library

* fix content args

* quest UI finally

* external navbar fix

* external navbar fix

* Add icons just to good view to present

* text crop fix

* lmao

* resolve problems

Co-authored-by: done_bd <dinil.bd@mail.ru>

Commit:
Fix meta repo usage

Commit:
Fix images in meta

AirBurn0 committed:
WIP quest content logic

AirBurn0 committed:
still WIP

Commit:
Implement quest storage

AirBurn0 committed:
fix background

AirBurn0 committed:
fix background

donebd committed:
fix json file

donebd committed:
Объединил костыльно две репы для верстальщика

donebd committed:
rename QuestItemRepo + minor clean + продолжение костыля с двумя репами

AirBurn0 committed:
UI minor fix, button colors etc

AlexanderKiselyov committed:
Create main.yml

AlexanderKiselyov committed:
Update main.yml

mariohuq committed:
Rename questMetaRepo, questContentRepo, (#24)

enable adding from two repos in one place in code, change addAll in metaRepo's, refactor readMetas, fix arguments!!

Co-authored-by: mariohuq <mariohuq@github.com>

AlexanderKiselyov committed:
Update main.yml

donebd commented:
![image](https://user-images.githubusercontent.com/55058233/162529312-e0617668-b51a-4d1c-b20a-6ed05f3912a1.png)
добренько)

AlexanderKiselyov committed:
Update main.yml

donebd commented:
тесты проходят из за того что у нас нет конфига тестов да?

AlexanderKiselyov commented:
подправил

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update build.gradle

AlexanderKiselyov committed:
Update build.gradle

Commit:
Remove unused findByName,
make json field name consts private,  refactor json parsing, remove unused path, add JSONArray.asIterableOfJSONObjects

mariohuq commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,47 @@
+    
+      - uses: actions/checkout@v2
+      
+      - name: Set up JDK 11
```
А здесь нельзя заиспользовать результат предыдущего step?

AlexanderKiselyov replied:
степы выполняются параллельно, попробую настроить на последовательное исполнение плюс используя успешный билд, будут далее проходить тесты

mariohuq commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,47 @@
+        run: chmod +x gradlew
+    
+      - name: Run all tests
+        run: ./gradlew --no-build-cache cleanTest test
```
А здесь переиспользовать build-cache если был успешный build

AlexanderKiselyov replied:
тоже попробую

mariohuq reviewed (changes requested):
Можно ли настроить схему, при которой step test запускается только при упешном step build и переиспользует его результат?

AlexanderKiselyov commented:
проблема пока в том, что ни один тест пока на самом деле не прогоняется; в поиске решения

AlexanderKiselyov committed:
Update build.gradle

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update build.gradle

AlexanderKiselyov committed:
Update build.gradle

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update main.yml

AlexanderKiselyov committed:
Update build.gradle

AlexanderKiselyov committed:
Update build.gradle

AlexanderKiselyov committed:
Update build.gradle
